File: Arcade/Graphs/1_Kingdom_Roads/01_New_Road_System/main.py
mainimport numpy
def newRoadSystem(roadRegister):
    npRoadRegister = numpy.array(roadRegister)
    for i in range(len(roadRegister)):
        if numpy.count_nonzero(npRoadRegister[i,:]) != numpy.count_nonzero(npRoadRegister[:,i]): return False
    return True

File: Arcade/Graphs/1_Kingdom_Roads/01_New_Road_System/main.ts
function newRoadSystem(roadRegister: boolean[][]): boolean {
    let rows: number[] = new Array(roadRegister.length).fill(0);
    let columns: number[] = new Array(roadRegister[0].length).fill(0);
    for (let i: number = 0; i < roadRegister.length; i++) {
        for (let j: number = 0; j < roadRegister[i].length; j++) {
            if (roadRegister[i][j]) {
                rows[i]++;
                columns[j]++;
            }
        }
    }
    for (let i: number = 0; i < rows.length; i++) {
        if (rows[i] !== columns[i]) {
            return false;
        }
    }
    return true;
}
